### ISSUE TITLE
Syntax highlighting broken in 1.19.6

### ISSUE THREAD

**Issue by mxj4** [rescript-lang/rescript-vscode#803]
I installed pre-release version 1.19.6 and found the end of the `%%raw()` cannot be identified correctly, now everything starting from `%%raw(` share the same color:
![image](https://github.com/rescript-lang/rescript-vscode/assets/881147/6dc9b61f-15ae-4bfd-b413-2b77e60a910c)
This does not happen in 1.18.0:
![image](https://github.com/rescript-lang/rescript-vscode/assets/881147/16ade4b7-235b-426a-aa32-a187d24ca513)

**Comment by fhammerschmidt** [comment]
The problem here is that JS syntax highlighting encounters no end symbol like a newline or a `;`.

But this can be easily fixed in the following ways:

1. Use regular strings:
```rescript
%%raw("import './App.css'")
```
  
2. Terminate the embedded JS with a semicolon
```rescript
%%raw(`import './App.css';`)
```
    
3. Or use multiple lines
```rescript
%%raw(`
  import './App.css'
`)
```

TBH I did not really account for the single line syntax highlight when I implemented this feature, what should we do in that case? One workaround would be to just automatically format to one of the above suggestions, another one would be to only support JS syntax highlighting in multiline extension points.
